compio-driver/src/lib.rs

```diff
@@ -259,7 +259,8 @@ impl Proactor {
             // SAFETY: completed.
             Some(unsafe { op.into_inner() })
         } else {
-            self.driver.cancel(op);
+            self.driver
+                .cancel(&mut unsafe { Key::<dyn OpCode>::new_unchecked(op.user_data()) });
```

Comment:
Wrap this conversion into a `safe` function?

Reply:
It's never safe. There should be only one Key which ownes the lifetime of one RawOp. The other Keys are only for convenience.